### DISCUSSION TITLE
Issue Labels

### DISCUSSION THREAD

**Discussion by josephj11** [autokey/autokey#836] [General]
@Elliria It looks like you created a couple of new issue labels (something most of us have also done unilaterally). That seem reasonable, but since this affects everybody:

Maybe we should discuss new candidates for labels before adding them so we are in agreement and get the best possible labels.

When new labels are added, they should be announced (Gitter...) so people know they are available and so that any questions as to when a particular label is appropriate or what it means can be resolved. When I assign labels, I usually quickly search for the ones I already know I want and check them off. I might miss seeing a relevant new one.

It's probably a good idea to keep the number of labels relatively small so that they are easy to manage and each label is assigned to enough issues to make it a useful search criterion... (Is there a search predicate for labels? There probably is, but I haven't seen it yet.)

This becomes more important because there are known issues with renaming labels which are already in use making improvements more difficult.

Here are a couple of questions to illustrate this issue.

What does "Todo" signify? How is that different than the fact that the issue is open and hasn't been closed yet? All valid issues are something someone wants done or they wouldn't have filed them.

How is "outreach" related to "upstream"?

**Comment by Elliria** [comment]
No worries. It was a repository I had created for myself to keep track of what I intend to do and the status of any particular task at any given moment. I didn't intend for it to be a community thing that required discussion or adaptation, so I made it just the way I like it.

Since, as you pointed out, it affected others and I never meant it to, I took it down and have gone back to tracking this sort of stuff offline. My apologies for any confusion or disorder that it caused.

The **todo** label represented something I could do immediately without preparation. The **outreach** label represented a need to contact someone before anything else could happen. Each label represented the very next step, so I could get a quick visual idea of what needed to happen without having to open the issues.

I'm not sure how my **outreach** label compared to your **upstream** label. I guess it depends on what the **upstream** label means for you. They're probably similar. I just called it outreach because that's how we refer to that activity in the digital marketing world.

I think it's all about interpretation in the end. For me:
* I read an **upstream** label as: "This is somebody else's (upstream's) problem and not mine."
* I read an **outreach** label as: "I need to reach out to someone."

I suppose it's also worth noting that the **outreach** label can involve any contacts rather than just upstream ones.

**Reply by josephj11** [reply]
Since label options are displayed in ascending alphabetical order, I would be fine with adding some personal labels with names like  zz_elliria_todo or anything similar that sort at the end of the list and are obviously not for general use.

This might actually be a good idea. And, since they're (potentially) personal, these wouldn't require any preapproval/review.

If we do go with something like this, we should standardize what these labels should be prefixed with. That way, if someone leaves the project, etc., we could easily determine which personal labels were theirs so we could easily delete them if desired. The label creator would also have confidence in which labels were  "theirs" so they could delete or rename them without concerns about affecting anyone else.

I am also open to new "regular" labels as I'm sure our current selections are not exhaustive, but  would prefer that they get reviewed first.

BTW, I checked and the search predicate is `label:`, so it's easy to use them to find issues alone or in combination with other predicates.

**Comment by Elliria** [comment]
Although I can see that having personalized labels could be useful, they won't be necessary for me. I'll just use what's there and suggest changes or new ones if they seem to be needed.

Speaking of which, the **What type of issue is this?** item in the issue tracker's interface gives you a hint as to the label to apply to the issue and I noticed that it's missing a **Development** or similar choice. Would you like me to add that to that list?

Also, the **Choose one or more terms that describe this issue:** item in the issue tracker's interface has **development**, but not **maintenance** in the list of choices, and since you recently mentioned that as a potential label, I figured I'd suggest it as an addition to that list as well.

Where do you use **label:** to do a search?

**Reply by josephj11** [reply]
Development and maintenance are valid labels (although maintenance hasn't been added yet), but I think they are team-only labels.  If we had a bunch of developers with your level of access, then they might make sense as suggestions, but, as it is, they should only be assigned by someone knowlegeable in AutoKey source/internals.

IMO:
A development issue is one that arises when coding/testing the development branch. End users don't usually do that.

A maintenance issue is one that arises either when reviewing the existing code base and finding deficiencies or existing things that need improvement or when something external changes that impacts  how AutoKey works. Again, not something end users are involved with.

![Selection_087](https://user-images.githubusercontent.com/4546046/230744773-ec5c0964-f694-48a9-91cb-98add9a08761.png)

See the Search field on the line that starts with `Filters`.

See [searching](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for more than you ever wanted to know.

**Comment by Elliria** [comment]
Would you like me to choose **None** from the **What type of issue is this?** field when filling out an issue report that involves development or maintenance?

And thanks for the screenshot of where to find the search box. I had been trying to get it working by putting it into the search box at the upper left in the [AutoKey issue search page](https://github.com/autokey/autokey/search?type=issues) and I see what I did wrong. I was putting a space between the colon and the search term (which will not work). In your screenshot, there's no space. As soon as I tried it without a space on the page I linked, it worked there, too. Good to know.

**Reply by josephj11** [reply]
Whatever you think works best or we can add them to the form.

Sooner or later we have to up your access so you can set the actual labels and do merges ...
